woocommerce/woocommerce-gutenberg-products-block
# Store Customization > Schedule an action to update the patterns content whenever the business description is changed

## What

Schedules an action whenever the store owner changes their business description is updated in the AI settings screen (whenever the ['woo_ai_describe_store_description'](https://github.com/woocommerce/woocommerce/blob/9418072a472b53bd4519086bc1958145600c42ae/plugins/woo-ai/includes/class-woo-ai-settings.php#L39) option is updated) to update the patterns images and content.

Fixes #10659

## Why



This is a requirement to ensure the images and content in patterns are properly updated whenever the business description is changed.

## Testing Instructions



_Please consider any edge cases this change may have, and also other areas of the product this may impact._

1. Install and activate the `woo-ai` plugin (it is available within the [WooCommerce monorepo](https://github.com/woocommerce/woocommerce), under the plugins directory).
2. From your dashboard, head over to WooCommerce > Settings > Advanced > Features
3. Fill out a description for your business and save

<img width="1044" alt="Screenshot 2023-09-16 at 09 22 29" src="https://github.com/woocommerce/woocommerce-blocks/assets/15730971/da3e8fc8-9e23-49a8-90ff-06d217192ece">

5. Now head over to Tools > Scheduled Actions
6. Search for the woocommerce_update_patterns_content hook
7. Make sure the action was successfully scheduled:

<img width="1032" alt="Screenshot 2023-09-16 at 09 27 23" src="https://github.com/woocommerce/woocommerce-blocks/assets/15730971/38efee14-bfc4-486c-8dcc-373963562603">

8. If the status is still Pending when you check, click on Run and ensure it is successfully executed. If the status is already Complete, no action is required as the process already finished successfully.
9. Check your error log and make sure you don't see any PHP errors.


## Screenshots or screencast



## WooCommerce Visibility


Required:


## Checklist

Required:

Conditional:

## Changelog


N/A
